provider.go

```diff
@@ -332,7 +332,7 @@ func ValidateProviderAddress(raw string) error {
 		}
 	}
 
-	if !p.IsLegacy() {
+	if p.IsLegacy() {
```

Comment:
See https://github.com/search?q=ValidateProviderAddress&type=code&p=1 for known public usage, which is practically none